### ISSUE TITLE
regression: boost dependency with the python module not found

### ISSUE THREAD

**Issue by jon-turney** [mesonbuild/meson#6844]
**Describe the bug**

Regression in detecting the python boost module 

**To Reproduce**

Consider the following `meson.build` (extracted from tizonia, listed in Users.md)
```
project('boost', 'cpp')
boost_dep = dependency(
  'boost', modules: ['chrono', 'filesystem', 'system', 'thread', 'program_options', 'python3'],
  required: true, version: '>=1.54'
)
```

On Ubuntu 18.04, with the  `libboost-chrono-dev`, `libboost-filesystem-dev`, `libboost-program-options-dev`, `libboost-python-dev`, `libboost-thread-dev` packages all installed.  `cmake` is not installed.

with meson 0.54.0.rc1
```
The Meson build system
Version: 0.53.999
Source dir: /home/jon/src/meson/test cases/frameworks/99 more
Build dir: /home/jon/src/meson/test cases/frameworks/99 more/_build
Build type: native build
Project name: boost
Project version: undefined
C++ compiler for the host machine: c++ (gcc 7.5.0 "c++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0")
C++ linker for the host machine: c++ GNU ld.bfd 2.30
Host machine cpu family: x86_64
Host machine cpu: x86_64
Run-time dependency Boost (modules: chrono, filesystem, program_options, python3, system, thread) found: NO 

meson.build:2:0: ERROR: Dependency "boost" not found
```

with meson 0.53.0
```
The Meson build system
Version: 0.53.0
Source dir: /home/jon/src/meson/test cases/frameworks/99 more
Build dir: /home/jon/src/meson/test cases/frameworks/99 more/_build
Build type: native build
Project name: boost
Project version: undefined
C++ compiler for the host machine: c++ (gcc 7.5.0 "c++ (Ubuntu 7.5.0-3ubuntu1~18.04) 7.5.0")
C++ linker for the host machine: c++ GNU ld.bfd 2.30
Host machine cpu family: x86_64
Host machine cpu: x86_64
Run-time dependency Boost (chrono, filesystem, system, thread, program_options, python3) found: YES 1.65.1
Build targets in project: 0

Found ninja-1.8.2 at /usr/bin/ninja
```

bisection lands on commit 08224dafcba1b694fb624553e7d84deb565aae22

**Expected behavior**

* the python boost module should be found
* when a specific boost module cannot be found, that should be identifed in or close to the error, not just written to the log.

**system parameters**
* native build
* Ubuntu 18.04
* Python 3.6.9
* 0.53.0, and 0.54.0.rc1

**Comment by jon-turney** [comment]
CC @mensinda 

There appears to be some coverage of this in `test cases\frameworks\1 boost`, but I guess we aren't hitting that in CI because we always have cmake installed?

**Comment by mensinda** [comment]
Neither CMake nor pkg-config are used for finding boost. Also, boost python is really anoying to find because you have to provide the major *and* the minor version. See the [test meson.build](https://github.com/mesonbuild/meson/blob/master/test%20cases/frameworks/1%20boost/meson.build#L33).

So, just passing `python3` as a module, shouldn't have worked even before my boost rewrite.

Also, I will look into better logging which module wasn't found.

**Comment by mensinda** [comment]
Should be fixed with #6855

**Comment by jon-turney** [comment]
> Should be fixed with #6855

Confirmed.  Thanks.
